### ISSUE TITLE
Accessibility issues: Focus trapped in textarea, labels for form controls, focus order for "Create" should be last

### ISSUE THREAD

**Issue by bugnumber9** [PrivateBin/PrivateBin#1386]
**Did you use the FAQ section?**
- [v] Yes, I have read [the FAQ](https://github.com/PrivateBin/PrivateBin/wiki/FAQ) and I found no solution/answer there.

I noticed several accessibility issues.

1. On the initial page load, the textarea is focused automatically.
There's no way to escape from it using keyboard navigation - the Tab key keeps adding tab characters inside the textarea.

2. Only the "New" button and password field have visible focus outlines. Dropdowns, checkboxes, and the "Create" button don't.

3. The "Create" button comes before the textarea, this will make users with keyboard navigation (assuming the issue 1 above is fixed) tab through the whole top menu again to create a secret.

**Browser:**
All browsers

**PrivateBin version:**
1.7.4

I can reproduce this issue on <https://privatebin.net>: Yes

**Comment by elrido** [comment]
Thank you for the great feedback!

Reg.:
1. Of course those things were deliberate design decisions to ease use of the tool (immediately start filling in the paste contents) and to edit text content, especially source code snippets that may use tabs. I found this discussion online, where one of the suggestions is to introduce a mode-key (Esc or Ctrl+M are suggested) and a description explaining this, close to the text area, that lets one disable the tab-capture. Then when tabbing around and returning to the textarea, the mode might get automatically (Esc) or manually (Ctrl+M) reset, for tabs to be used again in the editor. https://stackoverflow.com/questions/53909477/how-to-handle-tabbing-for-accessibility-with-a-textarea-that-uses-the-tab-button
2. I assume this to be a default choice for those button types in bootstrap CSS - we are working on a new version of that template, using a newer bootstrap CSS library. Could you take a peek at an instance that uses this new theme and comment if it has the same issue? For example, this instance uses the new theme: https://snip.dssr.ch/
3. I would have assumed keyboard oriented users (and I myself use tab-navigation a lot, especially on forms) would be aware of using shift-tab to cycle back. That way the create button should be the first element to select (or one of the first ones, maybe the edit/preview tabs are first.

So I could see that with a mode-key to disable the tab-capture, the workflow could become as simple as open URL, land in the textarea, type away or Ctrl-V (insert/paste), disable capture (key to be defined), shift-tab to create, space or enter to create the paste.

**Comment by bugnumber9** [comment]
1. The mode-key approach is a good idea.
The ultimate goal is to get rid of the keyboard trap - https://www.w3.org/WAI/WCAG22/Understanding/no-keyboard-trap
I'd say that supporting 2 mode-keys (Esc and Ctrl+M) would be nice since some accessibility software may override the default keyboard behavior.

2. The new version of the template is much better in terms of focus outline.
It still needs labels for form controls (password and textarea).

3. Yeah, I'm aware of Shift+Tab.
It's more about the focus order - https://www.w3.org/WAI/WCAG22/Understanding/focus-order
Also, it's kinda general consensus that "final" action goes after the form - think of send/submit button in a contact form, etc.

Appreciate you paying attention to this!

**Comment by elrido** [comment]
Suggested change got merged into master in #1463
